packages/components/src/components/Link/index.tsx

```diff
@@ -1,20 +1,45 @@
 import React, { PureComponent } from 'react';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 import PropTypes from 'prop-types';
 import colors from '../../config/colors';
 
 const A = styled.a<Props>`
     text-decoration: none;
     cursor: pointer;
-    color: ${props => (props.isGray ? colors.TEXT_SECONDARY : colors.GREEN_PRIMARY)};
-
-    &:visited,
-    &:active,
-    &:hover {
-        text-decoration: underline;
-        color: ${props => (props.isGray ? colors.TEXT_SECONDARY : colors.GREEN_PRIMARY)};
-    }
+    ${props =>
+        props.isGray &&
+        css`
+            color: ${colors.TEXT_SECONDARY};
+            &:visited,
+            &:active,
+            &:hover {
+                text-decoration: underline;
+                color: ${colors.TEXT_SECONDARY};
+            }
+        `}
+    ${props =>
+        props.isGreen &&
+        css`
+            color: ${colors.GREEN_PRIMARY};
+            &:visited,
+            &:active,
+            &:hover {
+                text-decoration: underline;
+                color: ${colors.GREEN_PRIMARY};
+            }
+        `}
+    ${props =>
+        props.hasNoStyle &&
+        css`
+            color: ${colors.TEXT};
```

Comment:
I would use here `color: inherit;`
So it will get the text color from parent element.